File: src/main/java/com/saomc/saoui/communication/Communicator.java
package com.saomc.saoui.communication;

import be.bluexin.saouintw.packets.PacketPipeline;
import be.bluexin.saouintw.packets.server.SendCommand;
import net.minecraft.client.Minecraft;
import net.minecraft.entity.player.EntityPlayer;

/**
 * Part of saoui
 *
 * @author Bluexin
 */
public class Communicator {

    public static boolean supportsPackets;

    public static void send(CommandType cmd, EntityPlayer target, String... args) {
        if (supportsPackets) PacketPipeline.sendToServer(new SendCommand(cmd, target, args));
        else new Command(cmd, target, args).send(Minecraft.getMinecraft());
    }
}

File: src/main/java/com/saomc/saoui/communication/Communicator.kt
package com.saomc.saoui.communication

import be.bluexin.saouintw.packets.PacketPipeline
import be.bluexin.saouintw.packets.server.SendCommand
import net.minecraft.client.Minecraft
import net.minecraft.entity.player.EntityPlayer

/**
 * Part of saoui

 * @author Bluexin
 */
object Communicator {

    var supportsPackets: Boolean = false

    fun send(cmd: CommandType, target: EntityPlayer, vararg args: String) = if (supportsPackets)
        PacketPipeline.sendToServer(SendCommand(cmd, target, *args))
    else Command(cmd, target, *args).send(Minecraft.getMinecraft())
}
